static/app/components/replays/replayPlayer.tsx

```diff
@@ -0,0 +1,95 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Consumer as ReplayContextProvider} from 'sentry/components/replays/replayContext';
+
+interface Props {
+  className?: string;
+}
+
+type Dimensions = {height: number; width: number};
+type RootElem = null | HTMLDivElement;
+
+type RootProps = {
+  initRoot: (root: RootElem) => void;
+  videoDimensions: Dimensions;
+  className?: string;
+};
+
+function PlayerRoot({className, initRoot, videoDimensions}: RootProps) {
+  const windowEl = useRef<HTMLDivElement>(null);
+  const viewEl = useRef<HTMLDivElement>(null);
+
+  const [windowDimensions, setWindowDimensions] = useState<Dimensions>();
+
+  // Create the `rrweb` instance which creates an iframe inside `viewEl`
+  useEffect(() => initRoot(viewEl.current), [viewEl.current]);
+
+  // Read the initial width/height where the player will be inserted, to setup view transform/scaling
+  // Also listen for resize events and so we can update view transform/scaling
+  useEffect(() => {
+    const handleResize = () =>
+      setWindowDimensions(windowEl.current?.getBoundingClientRect());
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, [windowEl.current]);
+
+  // Update the scale of the view whenever dimensions have changed.
+  useEffect(() => {
+    if (viewEl.current) {
+      const scale = Math.min((windowDimensions?.width || 0) / videoDimensions.width, 1);
+      if (scale) {
+        viewEl.current.style.transform = `scale(${scale})`;
+        viewEl.current.style.width = `${videoDimensions.width * scale}px`;
+        viewEl.current.style.height = `${videoDimensions?.height * scale}px`;
+      }
+    }
+  }, [windowDimensions, videoDimensions]);
+
+  return (
+    <div ref={windowEl} data-test-id="replay-window">
+      <div ref={viewEl} data-test-id="replay-view" className={className} />
+    </div>
+  );
+}
+
+const StyledPlayerRoot = styled(PlayerRoot)`
+  /* Make sure the replayer fits inside it's container */
+  transform-origin: top left;
+
+  /* Fix the replayer layout so layers are stacked properly */
+  .replayer-mouse-tail {
+    position: absolute;
+    pointer-events: none;
+  }
+
+  /* Skin the player to match our aesthetic */
+  iframe {
+    border: none;
+  }
+  .replayer-mouse {
```

Comment:
I liked the glowy circle that we currently have, but nb

Reply:
Fair enough, I ripped it all out to highlight just the basics that were needed to get it working. 

I can bring the colors back and keep it separate from this core player, to make it easier for design to parse out what we've added.

Reply:
brought back all the mouse css